components/script/dom/abortsignal.rs

```diff
@@ -162,6 +163,11 @@ impl AbortSignal {
                 reason.set(self.abort_reason.get());
                 pipe.abort_with_reason(cx, global, reason.handle(), realm, can_gc);
             },
+            AbortAlgorithm::Fetch(fetch_context) => {
+                rooted!(in(*cx) let mut reason = UndefinedValue());
+                reason.set(self.abort_reason.get());
+                fetch_context.abort_fetch(reason.handle());
```

Comment:
```
error[E0596]: cannot borrow `*fetch_context` as mutable, as it is behind a `&` reference
   --> components/script/dom/abortsignal.rs:169:17
    |
169 |                 fetch_context.abort_fetch(reason.handle());
    |                 ^^^^^^^^^^^^^ `fetch_context` is a `&` reference, so the data it refers to cannot be borrowed as mutable
```
Again, why is this not the case for `PipeTo`, but it states it is a reference here? I think it has to do with the `&mut self` and I need to put `locally_aborted` as a `Cell<bool>`, should I not?

Reply:
With `Arc<Mutex<FetchContext>>` this is no longer a problem, since locking provides a mutable reference.